### ISSUE TITLE
Гост 7.0.11-2011 предполагает группировку

### ISSUE THREAD

**Issue by xcoce** [Darxor/CSL-GOST#1]
Добрый день
по госту нужно в каждой группе проводить сортировки по алфавиту. группы:
1. нормативные акты 2 книги 3 Статьи 4 иностранная литра 5 Интернет-источники
можно это реализовать как то на основе вашего стиля?
[Пример в списке литры](http://old.fa.ru/dep/ods/autorefs/Dissertations/Колесников%20(28.02.2014)%2067327fcd90ff200d58a96f17db7b15d5.pdf)

**Comment by Darxor** [comment]
В примере есть два момента:
1. Собственно уровни сортировки _(сначала тип, потом алфавит)_
2. Разбивка групп заголовками

Для первого пункта нужно добавить [sorting macros](http://docs.citationstyles.org/en/stable/specification.html#sorting-macros), в котором эксплицитно указать порядок сортировки для различных типов документов.

Со вторым пунктом всё гораздо сложнее, т.к. в спецификации такого функционала нет и как его прикостылить тоже не очень ясно.

**Comment by xcoce** [comment]
> Собственно` уровни сортировки (сначала тип, потом алфавит)
> Для первого пункта нужно добавить sorting macros, в котором эксплицитно указать порядок сортировки для различных типов документов.

Типы значит разграничивет? и сортирует по алфавиту каждый тип? на Mendeley попробую скоро

**Comment by Darxor** [comment]
Да, в CSL типы разграничиваются и есть возможность указывать уровни сортировки. 

Соответственно, чтобы сделать сортировку

- Тип
    - Алфавит
        - Дата издания

Понадобится код а-ля:

```xml
<macro name="genre-sort">
    <choose>
        <if type="...">
            <text value="001"/>
        </if>
        <else-if type="...">
            <text value="002"/>
       </else-if>
    </choose>
</macro>
...
<bibliography second-field-align="flush">
    <sort>
        <key macro="genre-sort"/>
        <key macro="author-cite"/>
        <key variable="issued"/>
    </sort>
    ...
</bibliography>
```

Типы можно посмотреть в [приложениях к спецификации](http://docs.citationstyles.org/en/stable/specification.html#appendix-iii-types)

Есть нюанс, о котором точно сходу не отвечу: влияет ли порядок тегов `<layout>` на порядок сортировки, насколько помню — нет. В стиле есть несколько `<layout>`  для корректной работы многоязычных библиографий.

**Comment by xcoce** [comment]
[ваш переделанный GOST_R_7011_numeric.csl с макросом код а-ля: ](http://7works.ru/darxor/GOST_R_7011_numeric.csl)

добавил поле "genre" . статьям прописываю "article" , книгам "book". 
в списке литературы всё по прежнему, по жанрам(типам) не распределяются

[Есть такая ветка, тут всё почти работает, но вот английские  источники никак не выделю в отдельный тип](https://github.com/romanraspopov/GOST-styles-for-Zotero/issues/1)

для справки: тип - это нормативный акт,  книга, статья, ин. яз. литра, ссылка в интернете.

**Comment by xcoce** [comment]
всё решилось, Вам спасибо

**Comment by xcoce** [comment]
[посмотрите как тут решилась эта проблема для Mendeley и Zotero](https://github.com/romanraspopov/GOST-styles-for-Zotero/issues/1)
1. Законы (Statute)
2. Книги (book) 
3. Статьи (journal article)
4. Иностранщина (Television Broadcast)
5. Веб ссылки (web page)

вот сам стиль:
[gost-r-7-0-5-2008-num-type-alphab.zip](https://github.com/irandom/docs/files/1953340/gost-r-7-0-5-2008-num-type-alphab.zip)

Данная сортировка по типу, алфавиту (начиная в каждом типе) соответствует ГОСТ 7.0.11-2011

Однако порядок указания авторов, и другой информации об источнике возможно не соответствует данному госту

**Comment by Kgosha89** [comment]
добрый день. Человек очень далекий от программирования, но возникла необходимость отредактировать этот стиль https://raw.githubusercontent.com/Darxor/CSL-GOST/master/GOST_R_7011_numeric.csl так чтобы вначале списка шла русская литература. Возможно ли это сделать?
